crates/red_knot/src/salsa_db/semantic.rs

```diff
@@ -0,0 +1,577 @@
+
+                for clause in &node.elif_else_clauses {
+                    if clause.test.is_some() {
+                        // This is an elif clause. Create a new branch node. Its predecessor is the
+                        // previous branch node, because we can only take one branch in an entire
+                        // if/elif/else chain, so if we take this branch, it can only be because we
+                        // didn't take the previous one.
+                        prior_branch = self.flow_graph_builder.add_branch(prior_branch);
+                        self.set_current_flow_node(prior_branch);
+                    } else {
+                        // This is an else clause. No need to create a branch node; there's no
+                        // branch here, if we haven't taken any previous branch, we definitely go
+                        // into the "else" clause.
+                        self.set_current_flow_node(prior_branch);
+                        last_branch_is_else = true;
+                    }
+                    self.visit_elif_else_clause(clause);
+                    // Update `post_prior_clause` to a new phi node joining the possibility that we
+                    // took any of the previous branches with the possibility that we took the one
+                    // just visited.
+                    post_prior_clause = self
+                        .flow_graph_builder
+                        .add_phi(self.current_flow_node(), post_prior_clause);
+                }
+
+                if !last_branch_is_else {
+                    // Final branch was not an "else", which means it's possible we took zero
+                    // branches in the entire if/elif chain, so we need one more phi node to join
+                    // the "no branches taken" possibility.
+                    post_prior_clause = self
+                        .flow_graph_builder
+                        .add_phi(post_prior_clause, prior_branch);
+                }
+
+                // Onward, with current flow node set to our final Phi node.
+                self.set_current_flow_node(post_prior_clause);
+            }
+            _ => {
+                ast::visitor::preorder::walk_stmt(self, stmt);
+            }
+        }
+    }
+}
+
+#[tracing::instrument(level = "debug", skip(db))]
+#[salsa::tracked(jar=Jar)]
+pub fn dependencies(db: &dyn Db, file: File) -> Arc<[ModuleName]> {
+    struct DependenciesVisitor<'a> {
+        db: &'a dyn Db,
+        resolved_module: Option<ResolvedModule>,
+        dependencies: Vec<ModuleName>,
+    }
+
+    // TODO support package imports
+    impl PreorderVisitor<'_> for DependenciesVisitor<'_> {
+        fn enter_node(&mut self, node: AnyNodeRef) -> TraversalSignal {
+            // Don't traverse into expressions
+            if node.is_expression() {
+                return TraversalSignal::Skip;
+            }
+
+            TraversalSignal::Traverse
+        }
```

Comment:
`visit_expr` is probably easier. 

I don't want to build this as part of the semantic indexing because it would hurt concurrency. We call `dependencies` very early in `check` to schedule dependencies as early as possible. We don't want to block the fan out until type inference is complete.